pkg/mtbroker/filter/filter_handler.go

```diff
@@ -218,8 +218,14 @@ func (h *Handler) send(ctx context.Context, writer http.ResponseWriter, headers
 	statusCode, err := writeResponse(ctx, writer, response, ttl, span)
 	if err != nil {
 		h.logger.Error("failed to write response", zap.Error(err))
+		// Ok, so writeResponse will return the HttpStatus of the function. That may have
+		// succeeded (200), but it may have returned a malformed event, so if the
+		// function succeeded, convert this to an StatusInternalServerError instead to indicate
+		// error.
+		if statusCode == 200 {
+			statusCode = http.StatusInternalServerError
```

Comment:
502 might be a better status code since it indicates the error originated outside the broker. Separating the metrics for broker-internal errors and subscriber errors (sent on [the next line down](https://github.com/knative/eventing/pull/3450/files#diff-a9cf39e105b4ea1643e25fc890eb8462R229)) seems like a good idea for debugging.

Reply:
Hm, interesting. I was trying to find a more specific / better error code but nothing struck out because I too felt that separating them would be goodness. I'm down to make the change if we can get consensus on that approach.

Reply:
changed to 502.